src/librbd/io/ImageRequestWQ.h

```diff
@@ -107,6 +108,25 @@ class ImageRequestWQ
   std::atomic<unsigned> m_io_blockers { 0 };
   std::atomic<unsigned> m_io_throttled { 0 };
 
+  typedef interval_set<uint64_t> ImageExtents;
+
+  ImageExtents m_in_flight_extents;
+
+  vector<ImageDispatchSpec<ImageCtxT>*> m_blocked_ios;
+  ceph::mutex m_ordering_lock;
+
+  bool block_overlapping_io(
```

Comment:
Nit: pass `ImageDispatchSpec` instead of individual params

Reply:
It's probably now not possible with the modified code, since this check is now being made before creating a request